modules/vmss/variables.tf

```diff
@@ -57,325 +230,343 @@ variable "interfaces" {
+            grain_window_minutes       = 5
+            aggregation_window_minutes = 30
+            cooldown_window_minutes    = 60
+          }
+          scale_in_config = {
+            threshold               = 40
+            cooldown_window_minutes = 120
+          }
+        },
+        {
+          name = "Outbound Flows"
+          scale_out_config = {
+            threshold                  = 500
+            grain_window_minutes       = 5
+            aggregation_window_minutes = 30
+            cooldown_window_minutes    = 60
+          }
+          scale_in_config = {
+            threshold               = 400
+            cooldown_window_minutes = 60
+          }
+        }
+      ]
+    },
+  ]
   ```
   EOF
-  default     = null
-  type        = string
-}
-
-variable "autoscale_count_default" {
-  description = "The minimum number of instances that should be present in the scale set when the autoscaling engine cannot read the metrics or is otherwise unable to compare the metrics to the thresholds."
-  default     = 2
-  type        = number
-  nullable    = false
-}
-
-variable "autoscale_count_minimum" {
-  description = "The minimum number of instances that should be present in the scale set."
-  default     = 2
-  type        = number
-  nullable    = false
-}
-
-variable "autoscale_count_maximum" {
-  description = "The maximum number of instances that should be present in the scale set."
-  default     = 5
-  type        = number
-  nullable    = false
-}
-
-variable "autoscale_notification_emails" {
-  description = "List of email addresses to notify about autoscaling events."
   default     = []
-  type        = list(string)
   nullable    = false
-}
-
-variable "autoscale_webhooks_uris" {
-  description = "Map where each key is an arbitrary identifier and each value is a webhook URI. The URIs receive autoscaling events."
-  default     = {}
-  type        = map(string)
-}
-
-variable "autoscale_metrics" {
-  description = <<-EOF
-  Map of objects, where each key is the metric name to be used for autoscaling.
-  Each value of the map has the attributes `scaleout_threshold` and `scalein_threshold`, which cause the instance count to grow by 1 when metrics are greater or equal, or decrease by 1 when lower or equal, respectively.
-  The thresholds are applied to results of metrics' aggregation over a time window.
-  Example:
-  ```
-  {
-    "DataPlaneCPUUtilizationPct" = {
-      scaleout_threshold = 80
-      scalein_threshold  = 20
-    }
-    "panSessionUtilization" = {
-      scaleout_threshold = 80
-      scalein_threshold  = 20
-    }
+  type = list(object({
+    name          = string
+    minimum_count = optional(number)
+    default_count = number
+    maximum_count = optional(number)
+    recurrence = optional(object({
+      timezone   = optional(string)
+      days       = list(string)
+      start_time = string
+      end_time   = string
+    }))
+    scale_rules = optional(list(object({
+      name = string
```

Comment:
What do you think about adding validation if `name` is from 10 possible metrics:
```
DataPlaneCPUUtilizationPct
DataPlanePacketBufferUtilization
panGPGatewayUtilizationPct
panGPGWUtilizationActiveTunnels
panSessionActive
panSessionConnectionsPerSecond
panSessionSslProxyUtilization
panSessionThroughputKbps
panSessionThroughputPps
panSessionUtilization
```
?

Reply:
but name can be also a name of the built-in metric, we would have to maintain that list and follow changes on Azure. I'm not a fan of that :)